util/argo/argo.go

```diff
@@ -135,6 +136,29 @@ func TestRepoWithKnownType(ctx context.Context, repoClient apiclient.RepoServerS
 	return err
 }
 
+// ValidatePath validates the path specified in application spec. It is checked to ensure that
+// it does not leave the bounds of the repo
+func ValidatePath(path string) error {
+	cleanPath := filepath.Clean(path)
+	if filepath.IsAbs(cleanPath) {
+		return fmt.Errorf("application path is absolute")
```

Comment:
I think we actually have logic elsewhere to interpret absolute paths as being relative to the repo root. I gotta double-check though.